docs/_extra/api-reference/hypothesis-v1.yaml

```diff
@@ -0,0 +1,123 @@
+  license:
+    name: BSD (2-Clause)
+    url: https://github.com/hypothesis/h/blob/master/LICENSE
+servers:
+  - url: https://api.hypothes.is/api
+
+# -----------------------------------------------------------------------------
+# Reusable tags for grouping operations
+# -----------------------------------------------------------------------------
+tags:
+  - name: general
+  - name: annotations
+  - name: groups
+  - name: users
+
+# -----------------------------------------------------------------------------
+# Reusable components
+# -----------------------------------------------------------------------------
+components:
+
+  # -------------------------
+  # Resusable parameters
+  # -------------------------
+  parameters:
+    GroupID:
+      name: id
+      in: path
+      required: true
+      schema:
+        description: >
+          Either an 8-character alphanumeric unique identifier generated by the service
+          or an authority-provided identifier
+        oneOf:
+          - type: string
+            description: 8-character alphanumeric group ID assigned by the service
+            example: 4j7Dko3R
+          - type: string
+            pattern: "group:[a-zA-Z0-9._\\-+!~*()']{1,1024}@.*$"
+            description: Unique identifier assigned by the group's owning authority
+            example: "group:338facc93@myauthority.net"
+    GroupExpand:
+      name: expand
+      in: query
+      required: false
+      description: >
+        One or more relations to expand for the group resource
+      schema:
+        type: array
+        items:
+          type: string
+          enum:
+            - organization
+            - scopes
+
+  # --------------------------
+  # Resusable security schemes
+  # --------------------------
+  securitySchemes:
+
+    AuthClient:
+      type: http
+      scheme: basic
+      description: >
+        An AuthClient is a special registered set of credentials associated with a
+        particular "authority", (typically formatted as a domain, e.g. `myauthority.com`).
+        Requests authenticated in this manner allow the retrieval and manipulation
+        of resources (e.g. users, groups) across an entire authority, not just for
+        a single authenticated user.
+
+        These credentials must be pre-registered with the service provider.
+
+    ApiKey:
+      type: http
+      scheme: bearer
+      description: >
+        An API key corresponds to a single user. Requests authenticated in this
+        manner may retrieve and manipulate resources based on the authorization
+        of the associated user.
+
+  # ------------------------------
+  # Resusable resource schemas
+  # ------------------------------
+  schemas:
+    Group:
+      $ref: './schemas/group.yaml#/Group'
+
+
+# -----------------------------------------------------------------------------
+# API operations
+# -----------------------------------------------------------------------------
+paths:
+  /groups/{id}:
```

Comment:
Do you have a plan for how operations might be grouped and split out into smaller files in future?

Reply:
For the description document we _deliver_, all of the operations will be included in the main file (openapi API description documents are singular/monolithic). 

If we find the operations unrolled like this to be cumbersome to maintain, we could do basically whatever we wanted if we introduced a build step somewhere here. What I'd like to do is get all the operations into the base file, optimized, organized and validated, and then evaluate whether it's worth the effort to introduce a more modular build process here.

The overall description document _should_ be briefer than our existing documentation. For example, we can externalize a bunch of our `components`, including `parameters`, etc.

Summary: I'd like to see how long this thing ends up with sorta industry-standard organization. And avoid premature optimization. But definitely make the thing easier to navigate than the current version.